### PR TITLE
Miscellaneous logs fix for garage

### PR DESCRIPTION
**Pull request by Tiny-DM**
## What type of PR is this.
1. [X] Bug
2. [ ] Change
3. [ ] Enhancement

### What have you changed and why?
Information:
Changed the line to update the garage interface to only work when there's someone in the garage interface, removing 2 startup errors and 1 error every time someone garaged a vehicle. All errors purely present in the logs with no gameplay impact

### Please specify which Issue this PR Resolves.
closes [this discord comment from John](https://discord.com/channels/449481990513754112/637869964107186199/1183578742598148187)

### Please verify the following and ensure all checks are completed.

1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [X] No
2. [ ] Yes (Please provide further detail below.)

### CODE REVIEW

File: A3A/addons/garage/Public/fn_addVehicle.sqf [hunk 1]
```diff
@@ -216,7 +216,10 @@ private _refreshCode = {
     } forEach _cats;
     call HR_GRG_fnc_updateVehicleCount;
 };
-[ _catsRequiringUpdate, _refreshCode ] remoteExecCall ["call", HR_GRG_Users];
+
+if !(isNil "HR_GRG_Users" || HR_GRG_Users isEqualTo[]) then {
```
Comment (rautamiekka) [inline A3A/addons/garage/Public/fn_addVehicle.sqf:220]:
```suggestion
if !(isNil "HR_GRG_Users" || {HR_GRG_Users isEqualTo []}) then {
```

File: A3A/addons/garage/Public/fn_addVehiclesByClass.sqf [hunk 2]
```diff
@@ -62,5 +62,9 @@ private _refreshCode = {
     } forEach _cats;
     call HR_GRG_fnc_updateVehicleCount;
 };
-[ _catsRequiringUpdate, _refreshCode ] remoteExecCall ["call", HR_GRG_Users];
+
+if !(isNil "HR_GRG_Users" || HR_GRG_Users isEqualTo[]) then {
```
Comment (rautamiekka) [inline A3A/addons/garage/Public/fn_addVehiclesByClass.sqf:66]:
```suggestion
if !(isNil "HR_GRG_Users" || {HR_GRG_Users isEqualTo []}) then {
```

**Review by rautamiekka — CHANGES_REQUESTED** [review]
Trying to check a nil against anything will break, so it's 1 of the cases where lazy eval is mandatory.

File: A3A/addons/garage/Public/fn_addVehicle.sqf (hunk 1 above)
Comment (Tiny-DM) [reply A3A/addons/garage/Public/fn_addVehicle.sqf]: Resolved x2

**Comment by jaj22** [comment]
I'd just pull the isNils personally. If HR_GRG_Users is nil at that point then the rest of the code isn't going to work either. Having the isNil in there makes it look like it's a valid possibility.

**Review by jaj22 — APPROVED** [review]
Would rather it didn't have the isNil statements but it'll work.

**Comment by Tiny-DM** [comment]
You guys are too picky :P That's my excuse for not writing it properly the first time. Regardless, the isNil's have been pulled for clarity.

**Comment by rautamiekka** [comment]
> I'd just pull the isNils personally. If HR_GRG_Users is nil at that point then the rest of the code isn't going to work either. Having the isNil in there makes it look like it's a valid possibility.

Good point.

**Pull request merged**